internal/subscriber/subscriber.go

```diff
@@ -207,7 +210,21 @@ func (s *Subscriber) acknowledge(req *AckMessage) {
 	}
 
 	if shouldCommit {
-		_, err := s.consumer.CommitByPartitionAndOffset(ctx, messagebroker.CommitOnTopicRequest{
+		offsetUpdated := true
+		registryOffset := strconv.Itoa(int(offsetToCommit) + 1)
+		offsetModel := offset.Model{
+			Topic:        s.subscription.Topic,
+			Subscription: s.subscription.ExtractedSubscriptionName,
+			Partition:    req.Partition,
+			LatestOffset: registryOffset,
+		}
+		err := s.offsetCore.SetOffset(ctx, &offsetModel)
```

Comment:
don't we need to do the same thing also in `subscriber.retry` when we are comitting on the primary topic?

Reply:
Currently retry topics are not covered since retry might be in an ambiguous state when we rollback. The idea is to resume from the last successful ACK and retry all messages including ones pushed to the retry topics.